actions/call_api.py

```diff
@@ -1,10 +1,22 @@
 from lib.actions import ZabbixBaseAction
 from zabbix.api import ZabbixAPI
 
+import six
+
 
 class CallAPI(ZabbixBaseAction):
     def run(self, api_method, token, **params):
         # Initialize client object to connect Zabbix server
+
+        # Python doesn't let you modify the length of a dict or list while iterating
+        iterate_params = params.copy()
+
+        # Look for Empty strings and None values, but allow False
+        # If None or '', remove it from params.
+        for key, value in six.iteritems(iterate_params):
+            if key in params and params[key] is None:
+                del params[key]
```

Comment:
These lines could be simplify using dictionary comprehension as below. And that is compatible with Python 2.7 and 3.x.
```Python
{k:v for (k,v) in params.items() if v is not None}
```

By using this, you could implement this processing more simple as following.
```Python
return self._call_api_method(self.client, api_method,
                             {k:v for k,v in params.items() if v is not None})
```

Reply:
This is a MUCH cleaner solution. thank you!